scripts/generate_settings.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+import yaml
+import jinja2
+import re
+from settings_list import SettingsList
+import os
+
+sbp_generate_path = "../libswiftnav/sbp_generate"
+myenviron = os.environ
+texinputs = myenviron["TEXINPUTS"]
+myenviron["TEXINPUTS"]=".:" + sbp_generate_path + ":" + texinputs
+
+settings = SettingsList("settings.yaml")
+groups = settings.sl_return_groups()
+
+LATEX_SUBS = (
+    (re.compile(r'\\'), r'\\textbackslash'),
+    (re.compile(r'([{}_#%&$])'), r'\\\1'),
+    (re.compile(r'~'), r'\~{}'),
+    #(re.compile(r'\^'), r'\^{}'),
+    (re.compile(r'_'), r'_'),
+    (re.compile(r'"'), r"''"),
+    (re.compile(r'\.\.\.+'), r'\\ldots'),
+)
+
+NO_UNDERSCORE = re.compile(r'_')
+
+# We sometimes need to remove underscores.
+# This will remove the latex safe underscore character and replace with a space
+def no_us(value):
+    newval = value
+    if newval:
+      newval = NO_UNDERSCORE.sub(' ', newval)
+    return newval
+
+def escape_tex(value):
+    newval = value
+    if newval:
+      for pattern, replacement in LATEX_SUBS:
+          newval = pattern.sub(replacement, newval)
+    return newval
+
+jenv = jinja2.Environment(
+    block_start_string = '((*',
+    block_end_string = '*))',
+    variable_start_string = '(((',
+    variable_end_string = ')))',
+    comment_start_string = '((=',
+    comment_end_string = '=))',
+    loader=jinja2.FileSystemLoader("./")
+)
+jenv.filters['escape_tex'] = escape_tex
+jenv.filters['no_us'] = no_us
+
+
+latex_template = jenv.get_template('settings_template.tex')
+with open("settings_out.tex", 'w') as f:
+  f.write(latex_template.render(groups=sorted(groups), setting=sorted(settings.list_of_dicts)))
+
+import subprocess
+
+subprocess.Popen(["pdflatex" , "settings_out.tex"], env=myenviron)
+subprocess.call(["mv" , "settings_out.pdf", "../docs/settings.pdf"])
+
```

Comment:
Agree after reading the settings doc that it is very firmware specific and should probably not be part of the SBP documentation. Should chat with Mark and Buro about how to reduce duplication. Perhaps we could still include the machinery to generate settings specifications in libsbp (as its kinda part of the protocol, maybe?) but put these specific settings definitions in the firmware?